lib/Mojo/IRC.pm

```diff
@@ -594,6 +645,41 @@ CHUNK:
   }
 }
 
+sub _write_and_wait {
```

Comment:
I think maybe it should just time out by default after 30 seconds or something? Not sure about the callback.